test/support/utils.rb

```diff
@@ -162,25 +151,40 @@ def run_test_scenario(cmd, repl_prompt)
 
     private
 
+    def kill_remote_debuggee
+      if defined?(@remote_debuggee_pid) && @remote_debuggee_pid
+        @remote_r.close
+        @remote_w.close
+        Process.kill(:KILL, @remote_debuggee_pid)
+        Process.waitpid(@remote_debuggee_pid)
+        @remote_debuggee_pid = nil
+      end
+    end
+
     # use this to start a debug session with the test program
     def manual_debug_code(program)
       print("[Starting a Debug Session with @#{caller.first}]\n")
       write_temp_file(strip_line_num(program))
-
-      require_relative "../../lib/debug/client"
-
-      socket_path = DEBUGGER__.create_unix_domain_socket_name
-      inject_lib_to_load_path
-      ENV["RUBY_DEBUG_SOCK_PATH"] = socket_path
-      pid = spawn("#{RUBY} -r debug/open #{temp_file_path}")
+      socket_path = setup_socket_remote_debuggee
 
       while !File.exist?(socket_path)
         sleep 0.1
       end
 
       DEBUGGER__::Client.new([socket_path]).connect
     ensure
-      Process.kill('QUIT', pid)
+      kill_remote_debuggee
+    end
+
+    def setup_socket_remote_debuggee
```

Comment:
`socket_remote_debuggee` is ambiguous (TCP/IP also uses socket).

how about `setup_unix_doman_socket_remote_debuggee`?

Reply:
👍 updated